addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -6,6 +6,37 @@ const dom = require('web.dom');
 const publicWidget = require('web.public.widget');
 const utils = require('web.utils');
 
+// TODO: In master return/export `SharedPopupWidget` too
+const SharedPopupWidget = publicWidget.Widget.extend({
+    selector: '.s_popup',
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start() {
+        // A popup element is composed of a `.s_popup` parent containing the
+        // actual `.modal` BS modal. Our internal logic and events are hidding
+        // and showing this inner `.modal` modal element without considering its
+        // `.s_popup` parent. It means that when the `.modal` is hidden, its
+        // `.s_popup` parent is not touched and kept visible.
+        // It might looks like it's not an issue as it would just be an empty
+        // element (its only child is hidden) but it leads to some issues as for
+        // instance on chrome this div will have a forced `height` due to its
+        // `contenteditable=true` attribute. It will result in a ugly white bar.
+        // tl;dr: this is keeping those 2 elements visibility synchronized.
+        this.$target.on('show.bs.modal.SharedPopupWidget', () => {
+            this.$target[0].classList.remove('d-none');
+        });
+        this.$target.on('hidden.bs.modal.SharedPopupWidget', () => {
+            this.$target[0].classList.add('d-none');
+        });
```

Comment:
You bind something in the start but don't unbind it in the destroy actually :grimacing: So after you enter edit mode, each event handler is probably called twice when a modal opens/closes... that does not break anything I guess but should still be changed :+1: 

Also this may be fragile actually: when you save a page with a modal opened, the modal is first closed before save... but this handle which adds the d-none class is called after the closing animation is done (as it should). This may be random, I would not be surprise that sometimes the snippet is sometimes saved with the d-none class and sometimes not. Maybe you should explicitly add the d-none class on save? Although I guess that saved with or without the class, it will still work, best be consistent I think.

Reply:
> You bind something in the start but don't unbind it in the destroy actually grimacing

Note: why not use the "events" property to declare those handlers actually?

Reply:
@rdeodoo 

> We really should think about creating a util method to bind events (when we can't use the events attribute ofc) that would unbind the event in destroy automatically IMO, this seems to be a frequent issue everyone is making, no?

No, it's very rare: what you want is to use the `events`  key which is exactly there for that.

> I did that, see [here](https://github.com/odoo/odoo/compare/72129cf7d5dfc19799a8ceb989c431970a45134f..ab27efd3f58dccba29581a6f27c5971e195e81aa), but it was not working, the event was never triggered

This is what should be investigated IMO, I don't see the reason why it would not work so it may be a bug worth checking... The system normally do exactly what you are doing: binding events using jquery's `on` etc :grimacing: 

> Sure, but not sure where would it be better? In the cleanForSave in the s_popup/options.js? Or the other possibility would be in the cleanForSave of the SnippetEditor [here](https://github.com/odoo/odoo/blame/331b9435c59936c604c50f46f448ca19929b79e3/addons/web_editor/static/src/js/editor/snippets.editor.js#L326) which is in charge of closing the popup when saving? With a check if this is a .s_popup?

No, in this code right here: in the `destroy` of this public widget IMO :+1: Kinda a corner case but technically, the snippet XML definition should have the d-none class... but that may hurt the drag and drop system so probably cannot be done even in master. But in the end, it's natural that this is part of the widget system: this is the one in charge of synchronizing the snippet div d-none class with the inner modal visibility.

Note: actually you also used `$target` in this public widget which is the deprecated/compatibility stuff. You should use `this.$el`/`this.el` only (but I'd really like to know why the `events` prop does not work first anyway)

Reply:
>> We really should think about creating a util method to bind events (when we can't use the events attribute ofc) that would unbind the event in destroy automatically IMO, this seems to be a frequent issue everyone is making, no?
>
>No, it's very rare: what you want is to use the events key which is exactly there for that.
>
>> I did that, see [here](https://github.com/odoo/odoo/compare/72129cf7d5dfc19799a8ceb989c431970a45134f..ab27efd3f58dccba29581a6f27c5971e195e81aa), but it was not working, the event was never triggered
>
> This is what should be investigated IMO, I don't see the reason why it would not work so it may be a bug worth checking... The system normally do exactly what you are doing: binding events using jquery's on etc grimacing

I investigated, everything works fine on my end with `events`, so I push-forced with that version